src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1418,6 +1451,45 @@ MonodroidRuntime::typemap_managed_to_java (MonoReflectionType *type, const uint8
 	return embeddedAssemblies.typemap_managed_to_java (type, mvid);
 }
 
+#if defined (WINDOWS)
+const char*
+MonodroidRuntime::get_my_location ()
+{
+	HMODULE hm = NULL;
+
+	if (GetModuleHandleEx (GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS |
+	                      GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+	                      (LPCSTR) &get_xamarin_android_msbuild_path, &hm) == 0) {
+		int ret = GetLastError ();
+		log_warn (LOG_DEFAULT, "GetModuleHandle failed, error = %d\n", ret);
+		return nullptr;
+	}
+
+	WCHAR path[MAX_PATH * 2];
+	if (GetModuleFileNameW (hm, path, sizeof(path)) == 0) {
+		int ret = GetLastError ();
+		log_warn (LOG_DEFAULT, "GetModuleFileName failed, error = %d\n", ret);
+		return nullptr;
+	}
+
+	PathRemoveFileSpecW (path);
```

Comment:
This *can* return an error: https://docs.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathremovefilespecw

No idea what that would *mean*, and the docs don't say how or why it can fail, but it does have a `BOOL` return type, which *presumably* we should be checking…

Reply:
The return value doesn't indicate an error. It tells us whether "file spec" (i.e. file name) was removed from the path or not. We don't really care so there's no point in checking that.